lib/modules/manager/helmfile/extract.ts

```diff
@@ -51,15 +52,16 @@ export async function extractPackageFile(
     );
     return null;
   }
+
   for (const doc of docs) {
     // Always check for repositories in the current document and override the existing ones if any (as YAML does)
     if (doc.repositories) {
-      registryAliases = {};
+      registryData = {};
```

Comment:
```suggestion
```
You are resetting `registryData` iteration, that way you will have only the data for that specific document again. 🤔

Reply:
expected if the current document is defining it's own repos

Reply:
But can't a document define repos and reference ones in a other document at the same time?

Reply:
I don't think so. if I remember correctly it's always overriding the previous.

Reply:
```
or (const doc of docs) {
    // Always check for repositories in the current document and override the existing ones if any (as YAML does)
```
it says in the comment